425 documents received updated dashboard feel and new forms

MeganGeissler opened the pull request:
Pull request supports #425 #1298 #999 #305 #80 #79 #78 #77 #67 

Consolidating docs received into one script and adding additional forms. Instructions: https://hennepin.sharepoint.com/:w:/r/teams/hs-economic-supports-hub/BlueZone_Script_Instructions/NOTES/NOTES%20-%20DOCUMENTS%20RECEIVED.docx?d=w1dce0cc33ca541f68855f406a63ab02b&csf=1&web=1&e=pt6bYj

Forms Supported
-Asset Form: Reads ACCT, SECU, CARS for footer month. Ability to update ACCT, SECU, CARS panels by desired month, separate dialog to captures DHS 6054 form details, and case notes.
-ATR: Captures details in dialog and case notes under its own header
-AREP: Reads AREP panel if exists and prefills dialog, updates AREP panel if arep_update checkbox is checked, case notes under its own header
-Change: Captures details in dialog and case notes 
-EVF: Captures details in dialog and case notes 
-Hospice: Searches case notes back 1 year for previous HOSP case note if exists and prefills dialog and case notes under its own header
-IAA & IAA-SSI: Captures details in dialog, if update_pben checked update PBEN panel, and case notes 
-LTC1503: Captures details in dialog, if update_faci  checked update FACI panel if not already at max faci panels, and case notes under its own header
-MOF: Captures details in dialog and case notes 
-MTAF:  Captures details in dialog and case notes. If Check if all docs rec’vd are associated with MTAF is checked header of case note will be MTAF Processed: [staus].
-PSN: Reads WREG/DISA panel if exists and prefills dialog, if update wreg/disa check panels update, and case notes. 
-Shelter: Reads MEMB, ADDR, HEST, SHEL and prefills dialog, if update addr/hest/shel checked panels will update if changes are made in dialog, and case notes 
-Special Diet: Reads Case/Curr, if MFIP/MSA is Active/Pending and if diet status is approved DIET panel updates, and case notes
-Other forms: Allows user to document forms received that are not built into this script. Captures details in dialog and case notes 

Case Note Details: The name of each form selected will become the case note header. The only exceptions are: ATR, AREP, Hospice, and LTC 1503 are case noted individually. For MTAF, if Check if all docs rec’vd are associated with MTAF is checked, then the header of the case note will be *** MTAF Processed: [status] *** instead of a list of the documents received.

C-Love commented:
I am doing some review of the functionality, so I am going to add notes here before completing a formal review. I probably won't finish review today so will pick it back up next week. 

I am noticing that the asset form 'Action Taken' field will fill up with periods:
![image](https://github.com/Hennepin-County/MAXIS-scripts/assets/13418322/d12d4bb8-4cba-4904-948a-239c38076b26)

Seems to happen when the dialog reloads or loads the asset information. The case note of it seems weird:
![image](https://github.com/Hennepin-County/MAXIS-scripts/assets/13418322/9b8a4afa-ebc5-4ff0-aa79-43481eeb95c8)

Second, if the the 6054 option was selected, it makes the dialog movement weird. Every time the dialog reloads, the 6054 dialog appears before the dialog selected. Once all the mandatory fields are completed, the dialog will finish even if a different button was pressed (ie, trying to view another form before pressing the 'Complete' button.

Looks like the outstanding verifications information is creating a separate case note. 

I am playing on training case 342059 if you want to see how the CASE/NOTEs are created for the tests I am completing.

MDR22 commented on notes/documents-received-testing.vbs:
```diff
@@ -0,0 +1,4705 @@
+			End If
+
+			If form_type_array(form_type_const, form_errors) = evf_form_name then 'Error handling for EVF Form 
+				IF IsDate(evf_date_received) = FALSE THEN evf_err_msg = evf_err_msg & vbCr & "* Enter a vaid Document Date."
```
typo - valid

MDR22 commented on notes/documents-received-testing.vbs:
```diff
@@ -0,0 +1,4705 @@
+					If trim(ltc_1503_FACI_1503) = "" Then ltc_1503_err_msg = ltc_1503_err_msg & vbCr & "-   Enter facility name"
+					If ltc_1503_length_of_stay = "" Then ltc_1503_err_msg = ltc_1503_err_msg & vbCr & "-   Select length of stay"
+					If ltc_1503_level_of_care = "" Then ltc_1503_err_msg = ltc_1503_err_msg & vbCr & "-   Select level of care"
+					If IsDate(ltc_1503_admit_date) = FALSE Then ltc_1503_err_msg = ltc_1503_err_msg & vbCr & "-   Enter vaid admission date"
```
typo - valid

MDR22 reviewed (approved):
Made a comment about some typos, but otherwise looks great to me! I had a few additional thoughts:
- As the dialog navigates the forms, it would be helpful if the button showed as being pressed (i.e. display as text rather than button). When selecting multiple forms, it was sometimes confusing where I was at in getting through the forms.
- Would you want to limit the number of forms? I imagine it probably wouldn't ever come up but when adding forms one-by-one they started to extend past the dialog and then the dialogs got wonky when navigating through the forms.

C-Love commented:
I think it might be beneficial to reset the selection droplist to blank after adding the form in the form selection dialog. This is definitely a nice to have. 

The add forms checkbox has the IAA text cut off slightly.

I am a little confused about the PBEN update functionality. It looks like the when PBEN was updated when I selected and IAA form and asked it to update PBEN, there were 2 RSDI lines entered. Typically we need a line for SSI and one for RSDI if both forms are submitted.

C-Love reviewed (changes requested):
There is still something weird happening with the actions_taken variable in the Account information. The script entered:
* Actions taken: Reviewed asset amounts,,,,,,,,

I'm wondering if there is a conflit with the trim functionality and the right(action_taken, 2) = ", ".

The PBEN update is also tricky.

MeganGeissler commented:
@C-Love & @MDR22 - thanks for the feedback! I incorporated everything except making the dialog dynamic in length I'm going to add that to my future iteration list. 

Edits include: 
-fixed comma error with asset form
-revised IAA form and pben updating to handle for both forms
-fixed typos 
-clears document selection after "add" button is selected, 
-enlarged form selection dialog
-form button will display as text if on said form dialog
-replaced the word "item" with "thing"
-added video button to inital dialog (need to update link once video is created)
-revised Diet form dialog

MeganGeissler commented:
@courtrightd Revisions based on feedback: 
-Revised ASSET-add cash functionality
-IAA-PBEN updating logic to only update each month once and to skip updating if already updated
-LTC- dialog and updating logic to only add FACI panel if facility name doesn't match any other panel in the month
-PSN- fset/abawd handling
-General Updates: added cancel btn to dialogs, revised resource btns, casenotes full doc name if only one doc received, fixed TIKLs, replaced video link, revised casenoting, slight revisions to AREP & IAA dialog

courtrightd commented on notes/documents-received-testing.vbs:
```diff
@@ -0,0 +1,5260 @@
+					EMReadScreen pben_line_5, 54, 12, 40
+					EMReadScreen pben_line_6, 54, 13, 40
+					pben_line_1 = trim(replace(pben_line_1, "_", ""))
+					pben_line_2 = trim(replace(pben_line_2, "_"
```
4764: the_asset isn't defined for the array before this point, needs to be indexed? or move this down into the next For the_asset loop

MDR22 commented:
[celebrate]     Mark Riegel reacted to your message:
________________________________
From: MeganGeissler ***@***.***>
Sent: Tuesday, August 13, 2024 3:20:48 PM
To: Hennepin-County/MAXIS-scripts ***@***.***>
Cc: Mark Riegel ***@***.***>; Mention ***@***.***>
Subject: [External] Re: [Hennepin-County/MAXIS-scripts] 425 documents received updated dashboard feel and new forms (PR #1894)


CAUTION: This email was sent from outside of Hennepin County. Unless you recognize the sender and know the content, do not click links or open attachments.

Merged #1894<https://github.com/Hennepin-County/MAXIS-scripts/pull/1894> into master.

—
Reply to this email directly, view it on GitHub<https://github.com/Hennepin-County/MAXIS-scripts/pull/1894#event-13865283923>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/A3Y4EVUHVUXDLCJ353GFDT3ZRIP5BAVCNFSM6AAAAABKJ3YTLWVHI2DSMVQWIX3LMV45UABCJFZXG5LFIV3GK3TUJZXXI2LGNFRWC5DJN5XDWMJTHA3DKMRYGM4TEMY>.
You are receiving this because you were mentioned.Message ID: ***@***.***>


Disclaimer: If you are not the intended recipient of this message, please immediately notify the sender of the transmission error and then promptly permanently delete this message from your computer system.
